src/sourmash/sketchcomparison.py

```
@@ -0,0 +1,138 @@
+        k1 = self.mh1.ksize
+        k2 = self.mh2.ksize
+        if k1 != k2:
+            raise TypeError(f"Error: Invalid Comparison, ksizes: {k1}, {k2}. Must compare sketches of the same ksize.")
+        self.ksize = self.mh1.ksize
+        m1 = self.mh1.moltype
+        m2= self.mh2.moltype
+        if m1 != m2:
+            raise TypeError(f"Error: Invalid Comparison, moltypes: {m1}, {m2}. Must compare sketches of the same moltype.")
+        self.moltype= self.mh1.moltype
+        # check num, scaled
+        if not any([(self.mh1.num and self.mh2.num), (self.mh1.scaled and self.mh2.scaled)]):
+            raise TypeError("Error: Both sketches must be 'num' or 'scaled'.")
+
+    def downsample_and_handle_ignore_abundance(self, cmp_num=None, cmp_scaled=None):
+        """
+        Downsample and/or flatten minhashes for comparison
+        """
+        if self.ignore_abundance:
+            self.mh1_cmp = self.mh1.flatten()
+            self.mh2_cmp = self.mh2.flatten()
+        else:
+            self.mh1_cmp = self.mh1
+            self.mh2_cmp = self.mh2
+        if cmp_scaled is not None:
+            self.mh1_cmp = self.mh1_cmp.downsample(scaled=cmp_scaled)
+            self.mh2_cmp = self.mh2_cmp.downsample(scaled=cmp_scaled)
+        elif cmp_num is not None:
+            self.mh1_cmp = self.mh1_cmp.downsample(num=cmp_num)
+            self.mh2_cmp = self.mh2_cmp.downsample(num=cmp_num)
+        else:
+            raise ValueError("Error: must pass in a comparison scaled or num value.")
+
+    @property
+    def intersect_mh(self):
+        # flatten and intersect
+        return self.mh1_cmp.flatten().intersection(self.mh2_cmp.flatten())
+
+    @property
+    def jaccard(self):
+        return self.mh1_cmp.jaccard(self.mh2_cmp)
+
+    @property
+    def angular_similarity(self):
+        return self.mh1_cmp.angular_similarity(self.mh2_cmp)
+        # do we want to shield against error here? Or let TypeError through?
```

should we punt this to an issue, or ??

---

idk - I actually think it is probably best just to let the TypeError through here, and then only call `angular_similarity` in `*Result` classes if we're using abundances. That way this won't fail silently (e.g. return `None`) if any of us start using sketch comparison classes directly...

but I can pop this in an issue if you want!

---

also it was previously a rust panic .. I figured a TypeError was better 😂